Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/PreserveOutputsTests.cs

```diff
@@ -519,7 +611,7 @@ public void PreservingProcessConsumingDynamicOutputTest()
             RunScheduler().AssertCacheMiss(preservingProcess.PipId, dynamicOutputProducer.PipId);
 
             // Delete dynamic output.
-            FileUtilities.DeleteDirectoryContents(ArtifactToString(outputDirectory), deleteRootDirectory: true);
+            //FileUtilities.DeleteDirectoryContents(ArtifactToString(outputDirectory), deleteRootDirectory: true);

```

Comment:
>//FileUtilities.DeleteDirectoryContents(ArtifactToString(outputDirectory), deleteRootDirectory: true); [](start = 12, length = 102)

It looks like this modifies the behavior of the test. Can you add it back in or did you changes cause some unintended consequences?

Reply:
CMIIW. I think the deletion of the outputDirectory there is meant to test file materialization.

---
In reply to: [287427160](https://github.com/microsoft/BuildXL/pull/369#discussion_r287427160) [](ancestors = 287427160)

Reply:
I enabled that line again. But added the cachemiss assert after that: 



---
In reply to: [287444400](https://github.com/microsoft/BuildXL/pull/369#discussion_r287444400) [](ancestors = 287444400,287427160)